processor/src/decorators/mod.rs

```diff
@@ -186,6 +187,64 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given a polynomial ( in NTT form ), over quadratic extension field, of domain size
+    /// power of 2, on stack top, this routine interpolates the polynomial using inverse
+    /// Number Theoretic Transform, producing the polynomial in its coefficient form, which
+    /// is written on advice tape.
+    ///
+    /// Note, we're fixing that blow-up factor is 8.
+    ///
+    /// Input stack state looks like
+    ///
+    /// `[poly_domain_size, begin_memory_address, ...]`
+    ///
+    /// - `poly_domain_size` must be a power of 2.
+    /// - Only starting memory address i.e. first code-word's address is provided on stack, consecutive
+    ///  memory addresses are expected to be holding remaining `poly_domain_size - 1` many code-words.
+    /// - Because blow-up factor of 8 is being used, resulting coefficient form polynomial should have
+    /// (poly_domain_size >> 3) -many coefficients.
+    ///
+    /// Final advice tape should look like
+    ///
+    /// `[coeff0, coeff1, ..., coeffn, ...]` | n = poly_domain_size >> 2
+    ///
+    /// Program which is requesting this non-deterministic computation should read `coeff0`
+    /// first i.e. `coeffn` should be seen at the very end.
+    fn inject_ext2_intt_result(&mut self) -> Result<(), ExecutionError> {
+        const blowup_factor: usize = 8;
+
+        let poly_len = self.stack.get(0).as_int();
+        let poly_addr = self.stack.get(1).as_int();
+
+        if (poly_len & (poly_len - 1)) != 0 {
+            return Err(ExecutionError::NonPowerOf2PolyDomainSize(poly_len));
+        }
+
+        let mut poly = Vec::with_capacity(poly_len);
+        for i in 0..(poly_len >> 1) {
+            let word = self.get_memory_value(self.system.ctx(), poly_addr + i);
+            if word.is_none() {
+                return Err(ExecutionError::InaccessibleMemoryAddress(poly_addr + i));
+            }
+
+            let word = word.unwrap();
+            poly.push(Ext2Element::new(word[0], word[1]));
+            poly.push(Ext2Element::new(word[2], word[3]));
+        }
+
+        let twiddles = winterfell::math::fft::get_inv_twiddles::<Ext2Element>(poly_len as usize);
+        winterfell::math::fft::interpolate_poly(&mut poly, &twiddles);
+
+        let poly_felts = Ext2Element::as_base_elements(&poly[..(poly_len / blowup_factor)]);
```

Comment:
I would probably combing this with the loop below as something like this:
```Rust
for element in Ext2Element::as_base_elements(&poly[..result_size]).rev() {
    self.advice.write_tape(i);
}
```

Reply:
Did something similar in e06a5d5d